lxd/db/cluster/secrets.go

```diff
@@ -301,3 +301,64 @@ func deleteSecretsByID(ctx context.Context, tx *sql.Tx, ids ...int) error {
 
 	return nil
 }
+
+// GetAllBearerIdentitySigningKeys returns a map of identity ID to token signing keys. It should only be used to refresh
+// the identity cache.
+func GetAllBearerIdentitySigningKeys(ctx context.Context, tx *sql.Tx) (map[int]AuthSecretValue, error) {
+	q := `SELECT entity_id, value FROM secrets WHERE entity_type = ? AND type = ?`
+
+	identityIDToSigningKey := make(map[int]AuthSecretValue)
+	scanFunc := func(scan func(dest ...any) error) error {
+		var identityID int
+		var value AuthSecretValue
+		err := scan(&identityID, &value)
+		if err != nil {
+			return err
+		}
+
+		identityIDToSigningKey[identityID] = value
+		return nil
+	}
+
+	err := query.Scan(ctx, tx, q, scanFunc, EntityType(entity.TypeIdentity), SecretTypeBearerSigningKey)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to get auth secrets: %w", err)
+	}
+
+	return identityIDToSigningKey, nil
+}
+
+// DeleteBearerIdentitySigningKey deletes any signing keys for the identity.
+func DeleteBearerIdentitySigningKey(ctx context.Context, tx *sql.Tx, identityID int) (int64, error) {
```

Comment:
I'm not sure about this pattern of returning rowsAffected for a delete function call.
Whats the expected use case here? Allow the caller to be able to detect if the entity existed to be deleted?
If so, why not return an api.StatusErr not found if RowsAffected <= 0, this would be more consistent with the errors we return when getting a row right?

Reply:
Not a blocker right now, but just stood out as a little odd to me.

Reply:
The docblock should at least state what the int64 return value is though as otherwise it requires the caller to inspect the implementation of the function to see what it does. This shouldn't be required.

Reply:
This was to be able to re-use the function in different contexts. For example:
- Token revocation
  - Return 404 if no rows deleted
  - Return 200 if 1 or more rows were deleted
- Token rotation
  - Don't check number of rows deleted.

Now that I think about it. This would be much clearer if the function did that logic itself and returned an appropriate api.StatusError. Will fix.